Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -603,6 +603,78 @@ public class GraphTraverser: GraphTraversing {
         )
     }
 
+    public func targetsWithExternalDependencies() -> Set<GraphTarget> {
+        Set(graph.dependencies.compactMap { fromDependency, toDependencies in
```

Comment:
I don't really follow why we go through `dependencies` to get targets with external ones.

Can't we do something like:
```swift
Set(allInternalTargets().filter {
  !directExternalTargetDependencies(for: $0).isEmpty // This method does not exist, can be a modification of the existing `directTargetDependencies` traverser method
})
```

Reply:
Yes, we can do that. I'll refactor it.